pkg/kubevirt/machine_factory.go

```diff
@@ -0,0 +1,45 @@
+package kubevirt
+
+//go:generate mockgen -source=./machine_factory.go -destination=./mock/machine_factory_generated.go -package=mock
+
+import (
+	gocontext "context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/cluster-api-provider-kubevirt/pkg/context"
+	"sigs.k8s.io/cluster-api-provider-kubevirt/pkg/ssh"
+
+	"github.com/pkg/errors"
+)
+
+// MachineInterface abstracts the functions that the kubevirt.machine interface implements.
+
+type MachineInterface interface {
+	Create(ctx gocontext.Context) error
+	Exists() bool
+	IsReady() bool
+	Address() string
+	SupportsCheckingIsBootstrapped() bool
+	IsBootstrapped() bool
+	GenerateProviderID() (string, error)
```

Comment:
Please add description comment to each function (you can copy the descriptions from the implementation object)
For example from here: https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/blob/c65f77c2130d943866efb495cdc7eb795f116bc5/pkg/kubevirt/machine.go#L88

Reply:
Fixed